LibraryServices/Services/Library.cs

```diff
@@ -0,0 +1,300 @@
+        public Book UpdateBook(int id, Book book)
+        {
+            Book bookToUpdate = _booksInLibrary.Find((Book) => Book.Id == id);
+            if (book != null)
+            {
+                bookToUpdate.Name = book.Name;
+                bookToUpdate.AuthorId = book.AuthorId;
+                bookToUpdate.Year = book.Year;
+            }
+            return bookToUpdate;
+        }
+
+        /// <summary>
+        /// Delete Author from Book
+        /// </summary>
+        /// <param name="bookId">Book's Id</param>
+        /// <returns>Updated Book without author</returns>
+        public Book DeleteAutorFromBook(int id)
+        {
+            Book bookToUpdate = _booksInLibrary.Find((Book) => Book.Id == id);
+            bookToUpdate.AuthorId = null;
+            return bookToUpdate;
+        }
+
+        /// <summary>
+        /// Add Author To Book
+        /// </summary>
+        /// <param name="bookId">Book's Id</param>
+        /// <param name="authorId">Author's Id</param>
+        /// <returns>New Book with added Author</returns>
+        public Book AddAuthorToBook(int bookId, int authorId)
+        {
+            Book bookToUpdate = _booksInLibrary.Find((Book) => Book.Id == bookId);
+            bookToUpdate.AuthorId = authorId;
+            return bookToUpdate;
+        }
+
+        /// <summary>
+        /// Adds the genre to book.
+        /// </summary>
+        /// <param name="book_id">The book identifier.</param>
+        /// <param name="genre_id">The genre identifier.</param>
+        /// <returns>Is added new link.</returns>
+        public bool AddGenreToBook(int book_id, int genre_id)
+        {
+            bool result = false;
+
+            if (GetBook(book_id) != null && _genres.Any((genre) => genre.Id == genre_id))
+            {
+                _bookAndGenresConnection.Add(new KeyValuePair<int, int>(book_id, genre_id));
+                result = true;
+            }
+            return result;
+        }
+
+        #endregion
+
+        #region AuthorRegion
+
+        /// <summary>
+        /// Return All Authors
+        /// </summary>
+        public IEnumerable<Author> GetAllAuthors()
+        {
+            foreach (Author author in _authors)
+            {
+                yield return author;
+            }
+        }
+
+        /// <summary>
+        /// Return Author by Id
+        /// </summary>
+        /// <param name="id">Author's Id</param>
+        /// <returns>Author by Id</returns>
+        public Author GetAuthor(int id)
+        {
+            return _authors.Find((Author) => Author.Id == id);
+        }
+
+        /// <summary>
+        /// Update Author
+        /// </summary>
+        /// <param name="id">Author's Id</param>
+        /// <param name="author">New Author</
+        public Author UpdateAuthor(int id, Author author)
+        {
+            Author authorToUpdate = _authors.Find((Author) => Author.Id == id);
+            if (authorToUpdate != null)
```

Comment:
I suppose you are not handling the case when authorToUpdate is null. And not using the result from this method.